manimlib/mobject/svg/labelled_string.py

```diff
@@ -89,31 +106,67 @@ def parse(self) -> None:
         self.external_specified_spans = self.get_external_specified_spans()
         self.specified_spans = self.get_specified_spans()
         self.label_span_list = self.get_label_span_list()
-        self.containing_labels_dict = self.get_containing_labels_dict()
-        self.has_predefined_colors = self.get_has_predefined_colors()
+        self.check_overlapping()
 
     # Toolkits
 
+    def get_substr(self, span: Span) -> str:
+        return self.string[slice(*span)]
+
+    def handle_regex_method(func):
+        def wrapper(self, pattern, pos=0, endpos=9223372036854775807):
```

Comment:
please add `@wraps(func)` between these two lines, where `wraps` is imported from `functools`. this is more standardized.